test/built-ins/Promise/any/iter-arg-is-false-reject.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019 Sergey Rubanov. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-promise.any
+description: >
+  Promise.any(false) rejects with TypeError.
+info: |
+  Promise.any ( iterable )
+
+  ...
+  3. Let iteratorRecord be GetIterator(iterable).
+  4. IfAbruptRejectPromise(iteratorRecord, promiseCapability).
+  ...
+
+  #sec-getiterator
+  GetIterator ( obj [ , hint [ , method ] ] )
+
+  ...
+  Let iterator be ? Call(method, obj).
+  If Type(iterator) is not Object, throw a TypeError exception.
+  ...
+features: [Promise.any, Symbol.iterator]
+flags: [async]
+---*/
+
+try {
+  Promise.any(false).then(function() {
+    $DONE('The promise should be rejected, but was resolved');
+  }, function(error) {
+    assert.sameValue(Object.getPrototypeOf(error), TypeError.prototype);
+    assert(error instanceof TypeError);
+  }).then($DONE, $DONE);
+} catch (error) {
+  $DONE(`The promise should be rejected, but threw an exception: ${error.message}`);
+}
```

Comment:
~~I don't think we need to wrap these in a try catch.~~

This is consistent with https://github.com/tc39/test262/pull/1496

I don't like it, but this ship has sailed this way. I'll see if I can improve this in a follow up.